pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -676,6 +691,15 @@ func (adc *attachDetachController) syncPVCByKey(logger klog.Logger, key string)
 func (adc *attachDetachController) processVolumesInUse(
 	logger klog.Logger, nodeName types.NodeName, volumesInUse []v1.UniqueVolumeName) {
 	logger.V(4).Info("processVolumesInUse for node", "node", klog.KRef("", string(nodeName)))
+	// determine if VolumesInUse changes exists
+	cachedVolumes, ok := adc.cacheVolumesInUse[nodeName]
+	if ok && time.Now().Sub(cachedVolumes.lastUpdate) < CacheVolumesInUseExpire {
+		if cachedVolumes.volumes.Len() == len(volumesInUse) && cachedVolumes.volumes.HasAll(volumesInUse...) {
+			logger.V(4).Info("processVolumesInUse skipped for node", "node", klog.KRef("", string(nodeName)))
+			return
+		}
+	}
+
 	for _, attachedVolume := range adc.actualStateOfWorld.GetAttachedVolumesForNode(nodeName) {
 		mounted := false
 		for _, volumeInUse := range volumesInUse {
```

Comment:
If this runs between the volume being attached, and being mounted, we won't update the mounted state of the volume until the cache expires. Is there a way to prevent that?

Reply:
volumesInUse of node status will be updated only if kubelet have mounted the volume, so a volume that is attached but not mounted yet will not be in volumesInUse and no mounted state will be missed. Is my judgement accurate?

Reply:
volumesInUse of node status will be updated only if kubelet have mounted the volume, so a volume that is attached but not mounted yet will not be in volumesInUse and no mounted state will be missed. Is my judgement accurate?